backend/inspirehep/records/marshmallow/literature/ui.py

```diff
@@ -97,6 +103,18 @@ def get_formatted_earliest_date(self, data):
             return missing
         return format_date(earliest_date)
 
+    def get_fulltext_links(self, data):
+        field_data = []
+        fields_to_include = {
+            "arxiv_eprints": ArxivFulltextLink(),
+            "external_system_identifiers": KekFulltextLink(),
+            "documents": InternalFulltextLink(),
+        }
```

Comment:
IMHO we shouldn't use schema in schema, why we just create three methods that handle this? i.e.

```python
def get_arxiv_fulltext_link():
   ....

def get_kek_fulltext_link():
   ....

def get_internal_fulltext_link():
   ....
```

Reply:
@drjova what's the advantage of this?

Reply:
we're not calling ``dump`` schema in schema and it's just what we need

Reply:
IMHO your sollution will be even more complex. Here we have full info about the data source (so it if it arxiv, kek or internal url) so we do not kave to check almost anything. Inside Serializer we don't, so this will require more checks like: if keys are called `description` and `value` or are they called `schema` and `value`. Additionally recognition of schema KEK vs arXiv will require more logic inside this simple serializer. So there will be quire complex logic for simple thing.
Also Moving this amount of logic so deep into serializers is really bad idea.

Reply:
it's the wrong way to use marshmallow, if you want to put the logic in a ``schema`` you can use ``Nested``, calling ``schema`` in ``schema`` it's wrong. Which means ``pre_dump`` you analyzing the date and you let marshmallow to handle the dumping.

Reply:
But wait. You want to serialize it without serializer? Just pure dict in here? Then why we use serializes at all? Is this really what we want?